### DISCUSSION TITLE
How to get workflowdefinitions from workflowregistry

### DISCUSSION THREAD

**Discussion by NiveditaGalwade** [elsa-workflows/elsa-core#4915] [Design & tech]
We have implemented custom workflow by implementing IWorkflow. We are trying to get workflowdefinitions for those workflows which are created using Programmatic way.

**Comment by glime-ai** [comment]
To retrieve workflow definitions for workflows created programmatically (i.e., by implementing `IWorkflow`), you would typically use the `IWorkflowRegistry` service provided by Elsa. This service allows you to look up workflow blueprints, which include both programmatically defined workflows and those created using the designer.

Here's a general approach to retrieving workflow definitions for programmatically created workflows:

1. Inject the `IWorkflowRegistry` service into your class where you need to access the workflow definitions.
2. Use the `IWorkflowRegistry` to find the workflow blueprint by its ID or name.
3. Once you have the blueprint, you can access its definition, which includes all the activities and properties that make up the workflow.

Here's an example in C#:

```csharp
public class MyService
{
    private readonly IWorkflowRegistry _workflowRegistry;

    public MyService(IWorkflowRegistry workflowRegistry)
    {
        _workflowRegistry = workflowRegistry;
    }

    public async Task<WorkflowBlueprint?> GetWorkflowDefinitionAsync(string workflowId)
    {
        // Get the workflow blueprint by ID
        var workflowBlueprint = await _workflowRegistry.FindAsync(workflowId, VersionOptions.Published);
        
        // If you need to get the workflow definition as JSON, you might need to serialize the blueprint
        // This step depends on how you want to use the workflow definition

        return workflowBlueprint;
    }
}
```

Please note that the `VersionOptions.Published` parameter is used to retrieve the published version of the workflow. You can adjust this based on whether you want the latest, published, or a specific version of the workflow.

If you need to serialize the workflow blueprint to JSON or another format, you may need to use a serializer that can handle the blueprint object structure.

Keep in mind that the exact method names and usage could vary depending on the version of Elsa you are using, and the above code is a general guideline. You should refer to the Elsa documentation and source code for the most accurate and up-to-date information.

**Comment by NiveditaGalwade** [comment]
It worked. Thank you for response
